### ISSUE TITLE
Backup not working anymore after repair when a remote volume is missing

### ISSUE THREAD

**Issue by mnaiman** [duplicati/duplicati#1678]
In 2.0.1.10- 2.0.1.10_canary_2016-03-31 Im unable to perform backup (in version 2.0.1.9 it was fine) on two systems. Run repair following errors were logged and repair finished.
```
Error: Remote file referenced as backupname-b3c953c1a3b274e5a9a79e9d4ead7e857.dblock.7z.aes, but not found in list, registering a missing remote file
Error: Terminated with error: Could not find file '\\server\path\backupname-b3c953c1a3b274e5a9a79e9d4ead7e857.dblock.7z.aes'.

```
Now Im unable to backup because during backup following error is thrown:
`System.Exception: The database was attempted repaired, but the repair did not complete. This database may be incomplete and the backup process cannot continue. You may delete the local database and attempt to repair it again`

Now Im unable to run backup. So I took version 2.0.1.9 and duplicati database I hade in this version before upgrade and backup is working fine, but Im unable to upgrade :(

I think it is linked to this: #1503 Improve handling after repair when a remote volume is missing

**Comment by ulicky** [comment]
What im really missing is some command to run full verification of remote backups, that everything readable and consistent that could be run on demand (or scheduled monthly or from commandline) and after that what kenkendk suggested in #1503 If possible, the repair process could re-create the missing volumes. Otherwise, it could suggest deleting the affected backup sets to bring back the backup to a consistent state. 

Now I have some inconsistencies with only option to completely wipe out remote storage and start from scratch with lose history. I would rather lost only some parts of history (backup sets) :)

**Comment by renestach** [comment]
The verification was meant to diagnose problems of the remote storage system. That is what repairs are meant for, too. Both were never developed to work around issues in Duplicati itself.

If you are using the command line, you can define how many files are checked in the verification. I think you can check all. I think it is a good idea to improve the user interface further so that users can select if they want to verify just a few files or all files in the backup.

**Comment by FootStark** [comment]
@ulicky  We are currently discussing and implementing some changes to better handle missing blocks / blockfiles. This might take some time, but we will report back. Until then, feel free to share your ideas on how to handle these cases to suit the user's (your!) needs, e.g. on backend-verifications.

**Comment by ulicky** [comment]
Thats good news. 
For this "bug": As I read from latests commits, version 10 references some orphan file found in metadata, but these file actually dont exists in remote storage, which makes backup in undesired state.

For ideas: 
1)Right now I wrote powershell script(can share), which decompress all files with aes extension and decrypt them using 7z. Then generate report which files are corrupted (undecryptable or undecompressible), but don't know what to do next, how to find out which backup dataset are affected.
2)There are hundreds things that can happened on remote side (transfer problems, power failures, data corruption, human error...) I don't care if I loose some performed backups in history, but I need to be sure, when disaster came I'm able to restore backup, so I miss some "Verify and repair remote location". First 100% check of all files and if needed repair to consistent state.

**Comment by FootStark** [comment]
For now, regarding 1), there's code in LocalDatabase.RemoveRemoteVolume that should clean all entries in the db linking to blocks in a volume (including the files in them). You may want to call that from your PowerScript.You can then use LocalDatabase.RegisterRemoteVolume to set the volume's state to Deleted. You can also try to just set the volume state to Deleting using RegisterRemoteVolume, I think Duplicati should cleanup itself then on next start / repair.

**Comment by ulicky** [comment]
I have failed. :-(
I need to stay on version 2.0.1.9 until some repair process for RemoteVolume will be in place. I could be commandline tool to delete missing blocks/blocks files and delete/repair invalid index files for these errors to get RemoteVolume to consistent state:

Failed to process file: Daily-20160201T203000Z.dlist.7z.aes
System.IO.InvalidDataException: Found invalid data while decoding.

Failed to process file: Daily-20160206T203000Z.dlist.7z.aes
Newtonsoft.Json.JsonReaderException: After parsing a value an unexpected 

Failed to process file: Daily-20160212T203000Z.dlist.7z.aes
System.IO.InvalidDataException: Invalid JSON, expected property "metasize"

Failed to process file: Daily-20160312T203000Z.dlist.7z.aes
System.IO.InvalidDataException: Invalid JSON, expected property "size"

Remote file referenced as Daily-b3c953c1a3b274e5a9a79e9d4ead7e857.dblock.7z.aes, but not found in list, registering a missing remote file
